evap/staff/views.py

```diff
@@ -1106,20 +1106,35 @@ def evaluation_textanswers(request, semester_id, evaluation_id):
 
     if view == 'quick':
         visited = request.session.get('review-visited', set())
+        skipped = request.session.get('review-skipped', set())
         visited.add(evaluation.pk)
-        next_evaluation = find_next_unreviewed_evaluation(semester, visited)
-        if not next_evaluation and len(visited) > 1:
+        next_evaluations = find_unreviewed_evaluations(semester, visited | skipped)
+        if not next_evaluations and (len(visited) > 1 or len(skipped) > 0):
             visited = {evaluation.pk}
-            next_evaluation = find_next_unreviewed_evaluation(semester, visited)
+            skipped = set()
+            request.session['review-skipped'] = skipped
+            next_evaluations = find_unreviewed_evaluations(semester, visited | skipped)
         request.session['review-visited'] = visited
 
         sections = evaluation_sections + contributor_sections
-        template_data.update(dict(sections=sections, next_evaluation=next_evaluation))
+        template_data.update(dict(sections=sections, next_evaluations=next_evaluations))
         return render(request, "staff_evaluation_textanswers_quick.html", template_data)
 
     template_data.update(dict(evaluation_sections=evaluation_sections, contributor_sections=contributor_sections))
     return render(request, "staff_evaluation_textanswers_full.html", template_data)
 
+@reviewer_required
+def evaluation_textanswers_skip(request):
+    evaluation_id = request.POST["evaluation_id"]
+    evaluation = get_object_or_404(Evaluation, id=evaluation_id)
+    if evaluation.is_fully_reviewed:
+        raise PermissionDenied
```

Comment:
Why is this required?

Reply:
I will remove this. Can I remove the two lines above as well?